Assets/Script/Components/PickUp.cs

```diff
@@ -1,31 +1,27 @@
 using UnityEngine;
-
-public class PickUp : MonoBehaviour
+public abstract class PickUp : MonoBehaviour
 {
-    //renderer
     private Mesh model;
-    [SerializeField] private int lifeSpawn;
-
-    public float Health { get; private set; }
-    public int Bolts { get; private set; }
-    public int Ammo { get; private set; }
-
+    protected Rigidbody rb;
+    [SerializeField] protected int lifeSpawn;
+    [SerializeField] protected float amount;
+   
     private void Awake()
     {
-        Setup();
         Destroy(gameObject, lifeSpawn);
     }
 
-    private void Setup()
+    public void Init(Vector3 direction)
     {
-
+        TryGetComponent<Rigidbody>(out rb);
+        rb.AddForce(direction, ForceMode.VelocityChange);
```

Comment:
You are using TryGetComponent but not checking the result in case the component doesn't exist. If it's guaranteed to exist add a Serializable Field and avoid using GetComponent.

If it might not exist check the result of TryGetComponent before using rb:

```
if(TryGetComponent<RigidBody>(out rb)
{
     rb.AddForce(direction, ForceMode.VelocityChange);
}
```